proposals/0173-Read-Generic-Network-Signal-data.md

```diff
@@ -16,277 +16,306 @@ To enable OEM proprietary mobile apps to read raw CAN/Network data by specifying
+* OEMDataRef - This is a reference for OEM mapping table which defines signal attributes for this vehicle data items. OEMs may use this table to differentiate between various vehicle and SW configurations. SDL will pass along this reference to HMI, HMI would be responsible to resolve this reference using OEM specific mapping table.
 
-    <param name="id" type="String" maxlength="100" mandatory="true">
-		<description>
-			Parameter ID for the requested vehicle data.	
-			Should not be Blank or contain whitespaces.
-			Same ID can be used for GET/SUBSCRIBE/UNSUBSCIRBE requests for same vehicle data.
-			Refer to OEM specific vehicle data identifier list for information on format of identifier and mapping of vehicle data.
-		</description>
-    </param>
+* _array_ : To speciify if the vehicle data item/param response is an array.
+* _mandatory_ : To specify if the vehicle data param is mandatory to be included in response for overall vehicle data item.
 
-	<param name="networkDataRequestMode" type="NetworkDataRequestMode" mandatory="true">
-		<description>
-			Enumeration to control the network data request type.
-			GET
-			SUBSCRIBE
-			UNSUBSCRIBE
-		</description>
-    </param>
-	
-   <param name="networkDataAttribute" type="String" maxlength="500" minsize="1" maxsize="100" array="true" mandatory="false">
-		<description>
-			String Array of additional Network Data Attributes which identify network data as per OEM implementation 			
-		</description>
-    </param>
-	
-</function>
+* _id_ is required for top level vehicle data items while _dataType_ & _reference_ are required fields for vehicle data and sub-params, _array_ and _mandatory_ can be omitted. If omitted, _array_ defaults to *false* and _mandatory_ defaults to *true*
 
-<enum name="NetworkDataRequestMode">
-	<description>Describes how the GenericNetworkData should behave on the platform</description>
-	<element name="GET">
-		<description>Denotes that the request type is single read request for network data</description>
-	</element>
-	<element name="SUBSCRIBE" >
-		<description>Denotes that the request type is to receive network data update with every change
-		Request is ignored if requested network data is already subscribed 		
-		</description>
-	</element>
-	<element name="UNSUBSCRIBE" >
-		<description>Denotes that the request type is to end the previous network data subscription
-		Request is ignored if requested network data is NOT already subscribed
-		</description>
-	</element>
-</enum>
+#### Schema file location and updates
+We will keep a base JSON schema file for SDL core and proxy locally and another one in the OEM cloud. OEMs can update the schema file with OEM specific items which SDL core would be able to download over the cloud along with policy update and SDL proxy would be able to download via a URL passed to the app from HMI as an _endpoint_ shown below:
```

Comment:
Change to:
**A** base JSON schema file for SDL core and proxy **will be kept** locally and **an additional** one in the OEM cloud. OEMs can update the schema file with OEM specific **items, which** SDL core would be able to download over the cloud along with policy **updates**; SDL proxy would be able to download via a URL passed to the app from HMI as an _endpoint_ shown below: